oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/tool/AzureCheck.java

```diff
@@ -388,7 +389,13 @@ public int run() {
             persistence = ToolUtils.decorateWithCache(persistence, persistentCachePath, persistentCacheSizeGb);
         }
 
-        FileStoreBuilder builder = fileStoreBuilder(Files.createTempDir()).withCustomPersistence(persistence);
+        FileStoreBuilder builder = null;
+        try {
+            builder = fileStoreBuilder(Files.createTempDirectory(getClass().getSimpleName() + "-").toFile()).withCustomPersistence(persistence);
+        } catch (IOException e) {
+            e.printStackTrace(err);
+            return 1;
+        }
```

Comment:
Maybe throw an IllegalStateException instead of returning 1 to keep closest to the Guava behavior?

Reply:
I don't understand, the Guava version may also fail, just with a RuntimeException.